src/Logging/Logging/LogServiceFactory.cs

```diff
@@ -29,36 +37,45 @@ public ILogService Create(string context)
             if (_services.TryGetValue(context, out var service))
                 return service;
 
-            service = new LogServiceComposition(
-                CreateContextAwareLogServiceProviders(context));
-
+            service = new LogService(CreateScopedLogServices(context));
             _services.TryAdd(context, service);
 
             return service;
         }
 
+        /// <summary>
+        /// Method for adding a <see cref="ILogServiceProvider"/> to the <see cref="LogService"/>.
+        /// </summary>
+        /// <param name="_this"></param>
         public void AddProvider<TProvider>(TProvider provider)
             where TProvider : class, ILogServiceProvider
         {
             Argument.NotNull(
                 provider,
                 nameof(provider));
 
+            var providerType = typeof(TProvider);
+            
+            if (!_providerTypes.Add(providerType))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5626772)